src/test/scala/org/broadinstitute/dsde/agora/server/webservice/EntityCreationPermissionSpec.scala

```diff
@@ -0,0 +1,189 @@
+
+package org.broadinstitute.dsde.agora.server.webservice
+
+import java.util.UUID
+
+import org.broadinstitute.dsde.agora.server.AgoraTestData._
+import org.broadinstitute.dsde.agora.server.dataaccess.permissions.AgoraPermissions.All
+import org.broadinstitute.dsde.agora.server.dataaccess.permissions.{AccessControl, AgoraPermissions}
+import org.broadinstitute.dsde.agora.server.model.AgoraApiJsonSupport._
+import org.broadinstitute.dsde.agora.server.model.{AgoraEntity, AgoraEntityType}
+import org.broadinstitute.dsde.agora.server.webservice.routes.MockAgoraDirectives
+import org.broadinstitute.dsde.agora.server.webservice.util.ApiUtil
+import org.scalatest.DoNotDiscover
+import spray.http.StatusCodes._
+import spray.http.StatusCode
+import spray.httpx.SprayJsonSupport._
+import spray.httpx.unmarshalling._
+
+@DoNotDiscover
+class EntityCreationPermissionSpec extends ApiServiceSpec {
+
+  var testEntity1WithId: AgoraEntity = _
+  var testEntity2WithId: AgoraEntity = _
+  var testEntity3WithId: AgoraEntity = _
+  var testEntity4WithId: AgoraEntity = _
+  var testEntity5WithId: AgoraEntity = _
+  var testEntity6WithId: AgoraEntity = _
+  var testEntity7WithId: AgoraEntity = _
+  var testEntityToBeRedactedWithId: AgoraEntity = _
+
+  override def beforeAll() = {
+    ensureDatabasesAreRunning()
+    // create namespace1, owned by owner1; namespace creation is a side effect of entity creation
+    testEntity1WithId = patiently(agoraBusiness.insert(testEntity1, owner1.get))
+    // create namespace2, owned by owner2; namespace creation is a side effect of entity creation
+    testEntity2WithId = patiently(agoraBusiness.insert(testEntity2, owner2.get))
+    // add owner3 as owner on namespace2
+    patiently(permissionBusiness.insertNamespacePermission(AgoraEntity(testEntity2.namespace), owner2.get, AccessControl(owner3.get, AgoraPermissions(All))))
+  }
+
+  override def afterAll() = {
+    clearDatabases()
+  }
+
+  behavior of "Agora, when creating/editing entities (methods and configs)"
+
+  it should "set up test fixtures correctly" in {
+    val ns1perms = patiently(permissionBusiness.listNamespacePermissions(AgoraEntity(testEntity1.namespace), owner1.get))
+
+    val expectedNs1Perms = List(AccessControl(owner1.get, AgoraPermissions(All)))
+
+    assertResult(expectedNs1Perms) {ns1perms}
+
+    val ns2perms = patiently(permissionBusiness.listNamespacePermissions(AgoraEntity(testEntity2.namespace), owner2.get))
+
+    val expectedNs2Perms = List(
+      AccessControl(owner2.get, AgoraPermissions(All)),
+      AccessControl(owner3.get, AgoraPermissions(All))
+    )
+
+    assertResult(expectedNs2Perms) {ns2perms}
+  }
+
+  it should "allow creating a new entity if you have permissions on the namespace" in {
+    // owner1 create in namespace1
+    assertEntityCreation(owner1.get, namespace1.get, randUUID)
+  }
+
+  it should "reject creating a new entity if you lack permissions on the namespace" in {
```

Comment:
Would it be helpful to also test rejection if `owner3` tries to create an entity on `namespace1`? It's slightly different coverage, because `owner3` has never created an entity before while `owner1` has